src/client/gl-texture.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "client/gl-buffer.h"
+#include "client/resource.h"
+#include "core/common.h"
+
+namespace zen::remote::client {
+
+class AtomicCommandQueue;
+
+enum class TextureTarget {
+  kNone,
+  kImage2D,
+};
```

Comment:
This seems better to be private within GlTexture.

And the name "target" seems a little bit misleading as we come up with something like `GL_TEXTURE_2D`.

Combining two things above, how about creating `enum GlTexture::ImageType` with its entry `kNone` and k2D`?